subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1295,7 +1295,7 @@ private void rethrowFailure(String type, Collection<Throwable> failures) {
 
     private void assertIsResolvable() {
         if (!canBeResolved) {
-            throw new IllegalStateException("Resolving configuration '" + name + "' directly is not allowed");
+            throw new IllegalStateException("Resolving dependency configuration '" + name + "' on its own is not allowed");
```

Comment:
:o: I am not sure if "on its own" is much clearer than "directly" (although in general it is better).
I think having a concrete suggestion in the message here would be more helpful for users. Something like:
`Resolving dependency configuration 'internal' is not allowed as it is defined as canBeResolved=false. Instead, a resolvable (canBeResolved=true) dependency configuration that extends 'internal' should be resolved.`